test/funder.js

```diff
@@ -0,0 +1,328 @@
+/**
+ * @typedef {Object} FundingReceipts
+ *
+ * @property {Array<Object>} originFaucetReceipts Array of receipts for
+ *                                                origin chain addresses funded
+ *                                                from mosaic faucet.
+ * @property {Array<Object>} auxiliaryFaucetReceipts Array of receipts for
+ *                                                   auxiliary chain addresses
+ *                                                   funded from mosaic faucet.
+ * @property {Array<Object>} ropstenFaucetReceipts Array of receipts for
+ *                                                 origin chain addresses funded
+ *                                                 from ropsten faucet
+ */
+
+/**
+ * @typedef {Object} FundingTransactionHashes
+ *
+ * @property {Array<string>} originFaucetTXHashes Array of transaction
+ *                                                hashes for origin chain
+ *                                                addresses funded from
+ *                                                mosaic faucet.
+ * @property {Array<string>} auxiliaryFaucetTXHashes Array of transaction
+ *                                                   hashes for auxiliary
+ *                                                   chain addresses funded
+ *                                                   from mosaic faucet.
+ * @property {Array<string>} ropstenFaucetTXHashes Array of transaction hashes
```

Comment:
The distinction `origin`, `auxiliary`, `ropsten` doesn't really make sens.

1. `origin` *is* `ropsten` in our first case
2. the test should not be bound to `ropsten`

Reply:
I think it makes sense in the context of the faucet. 
There are two faucet here: 
1. Ropsten faucet 
2. Mosaic faucet 

Ropsten faucet funds eth on ropsten and mosaic faucet funds ost on the origin and auxiliary chain. 

It's true that tests do not depend on chains, however, funding is chain specific.